src/runtime/virtcontainers/kata_agent.go

```diff
@@ -379,6 +381,43 @@ func (k *kataAgent) internalConfigure(h hypervisor, id string, config interface{
 	return nil
 }
 
+func setupSandboxBindMounts(sandbox *Sandbox) error {
+	if len(sandbox.config.SandboxBindMounts) == 0 {
+		return nil
+	}
+
+	// Create subdirectory in host shared path for sandbox mounts
+	sandboxMountDir := filepath.Join(getMountPath(sandbox.id), sandboxMountsDir)
+	if err := os.MkdirAll(sandboxMountDir, DirMode); err != nil {
+		return fmt.Errorf("Creating sandbox shared mount directory: %v: %w", sandboxMountDir, err)
+	}
+
+	for _, m := range sandbox.config.SandboxBindMounts {
+		mountDest := filepath.Join(sandboxMountDir, filepath.Base(m))
```

Comment:
It's not a good idea to use the host file's base as the mount destination's file name, since there may be two different host files with different file path but with the same file base, such as /dir1/test.txt and /dir2/test.txt, in such case, the second mounted file will override the first one.

Reply:
Ah, interesting. I see your point. I don’t expect this to be commonly adjusted and/or to utilize many files. I’m wondering if we should check for this and warn? I’m hesitant to making a unique path on behalf of the provider/user/operater, since I want the resulting path to be predictable for the guest service. 

What do you think makes sense, or suggest?

Reply:
Hi @egernst 

If you want to make it simple and make sure those sandbox mounted files base different, you can just check for it and return an error to users to rename the file?

Reply:
This makes sense to me. I’ll wait and see if others have input, but otherwise will add this check.  Thanks @lifupan!

Reply:
Done!